### DISCUSSION TITLE
Skeleton Roadmap to v1.0

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/skeleton#503] [Ideas]
> [!TIP]
> See our latest plans here:
> - https://github.com/skeletonlabs/skeleton/discussions/2375

---

The Skeleton core team recently met and gathered our ideas for what Skeleton v1.0 might look like. This document represents the key items for that goal. Please note that priority is not set in stone, things may change, but generally speaking the most critical items are listed towards the top. If it's not immediately clear, our focus is on quality and stability over quantity of features.

Have we missed something? Curious to learn more about an item listed here? Let us know in the comments below!

> NOTE: checked = released

## v1.0 Roadmap

- [x]  Refactors to Major Features:
    - [x] https://github.com/Brain-Bones/skeleton/issues/197
    - [x] https://github.com/Brain-Bones/skeleton/issues/198
    - [x] https://github.com/Brain-Bones/skeleton/issues/217
- [x]  Onboarding & Docs
    - [x]  https://github.com/Brain-Bones/skeleton/issues/512
    - [x]  https://github.com/Brain-Bones/skeleton/issues/552
- [x] https://github.com/Brain-Bones/skeleton/issues/465
- [x] https://github.com/skeletonlabs/skeleton/issues/528
- [x]  Design
    - [x]  https://github.com/Brain-Bones/skeleton/issues/151
    - [x] https://github.com/Brain-Bones/skeleton/issues/467
    - [x] https://github.com/Brain-Bones/skeleton/issues/553
    - [x]  https://github.com/skeletonlabs/skeleton/issues/598
    - [x]  https://github.com/skeletonlabs/skeleton/issues/493
- [x] https://github.com/Brain-Bones/skeleton/issues/389
- [x]  Recommended components (we'll use these in the Docs)
    - [x]  https://github.com/Brain-Bones/skeleton/issues/376 → needed for the theme generator
    - [x]  https://github.com/Brain-Bones/skeleton/issues/375 → requested multiple times, useful for docs
    - [x]  https://github.com/Brain-Bones/skeleton/issues/357 → used throughout the docs
- [x] Plugin & Bundle Optimizations
    - [x] https://github.com/skeletonlabs/skeleton/issues/650
- [x]  SEO
    - [x] https://github.com/Brain-Bones/skeleton/issues/355
    - [x] https://github.com/skeletonlabs/skeleton/issues/736
- [x]  Community & Support
    - [x]  https://github.com/Brain-Bones/skeleton/issues/551
    - [x]  https://github.com/skeletonlabs/skeleton/issues/780
    - [x]  https://github.com/skeletonlabs/skeleton/issues/781
- [x] Polish & Refinement
    - [x] https://github.com/skeletonlabs/skeleton/issues/837
    - [x] https://github.com/skeletonlabs/skeleton/issues/857
    - [x] https://github.com/skeletonlabs/skeleton/issues/922
    - [x] https://github.com/skeletonlabs/skeleton/issues/783

## Post v1.0

> **Note** the Skeleton core team will be meeting shortly after the v1.0 launch to regroup and evaluate and set expectations for upcoming releases. This will also define key goals leading up to the next major milestone release (v2.0)

- [ ]  Begin promoting to news sources and influencers
- [ ]  https://github.com/Brain-Bones/skeleton/issues/150
- [ ]  https://github.com/Brain-Bones/skeleton/issues/377 → needed for doc search
- [ ] https://github.com/skeletonlabs/skeleton/issues/732

### Proposals

- [ ] Improve project CI/CD
- [ ] Move to monorepo structure - if relevant
- [ ] Implement Changesets
- [ ] [Framework agnostic proposal](https://i.imgur.com/PwjPeyk.png)
- [ ] I18n support - RTL, data/time, numeric, plurals, etc (leave this to upstream?)
- [ ] Premium services - templates/support/etc
- [ ] [Husky Hooks](https://typicode.github.io/husky/#/) - to improve automation

**Comment by geoffkendall** [comment]
This looks excellent. One addition that would be great is to have more fully-functional menus, specifically to allow nesting (where specific menu items can themselves show sub-menus), and for menus to appear on hover rather than click. This will enable more desktop-like navigation of complex options. I don't think there's a way to do these things right now, but please correct me if I'm wrong! Thanks a lot,
Geoff

**Reply by endigo9740** [reply]
@geoffkendall, the proposals above are the "big stuff" to keep ourselves focused and organized. However, there's plenty of other small updates I anticipate we'll see over time. For example, your feature request for hover menus is open here:

- https://github.com/Brain-Bones/skeleton/issues/470

Just because it's not on the list above doesn't mean it's not something we'll get to sooner or later. Overall you might notice the goal is less "add a thousand features" and more "improve what we have". That way we can create a solid foundation to build off of, then new features can come in a much more rapid fashion.

If the basics aren't well established, then we'll end up with breaking changes over and over, which is the antithesis of stability. And stability is our north star when it comes to a full launch.

**Reply by geoffkendall** [reply]
Ah gotcha, thanks for the clarification! :-)

**Comment by 0x524c** [comment]
I'm loving using Skeleton. This looks excellent!! 🤩🚀

**Comment by ghost** [comment]
Based on the docs, this project looks pretty interesting. May I ask if there are any planned breaking changes for this library? I want to be a beta tester, but I don't know how stable Skeleton is...

**Reply by endigo9740** [reply]
Hey @IRod22, the project is currently in "pre-release" status per the homepage on our website. We've been working on the project for a year, open source since August, with new releases dropping every two weeks. This upcoming Tuesday will be our next release and the first "release candidate". Meaning the library itself will be stable enough that we can finally recommend it for production use.

Our v1.0 release is scheduled to come shortly after that (no firm ETA yet), but is dependent on updates such as this one being completed:
- https://github.com/skeletonlabs/skeleton/issues/783

If you're interested in contributing, I'd recommend reading through the ticket above, other tickets under Issues, and perhaps joining our Discord and browsing the #contributor channel - this should get you caught up on where we are on things. If you think you can help with any ongoing tasks let us know. The first we'll need is folks helping read through and fix issues in the documentation.

Also see our contribution guidelines here:
https://www.skeleton.dev/docs/contributions
